### PR TITLE
Update to Http4s 0.17.0-M2, Cats, and FS2

### PR DESCRIPTION
**Pull request by zarthross**
First crack at updating to Cats/Fs2.

### CODE REVIEW

**Comment by rossabaker** [comment]
Neat.  What's going to be the versioning strategy for rho?  Is it going to skip over http4s-0.16, or should he minor version be bumped out two versions?

(For those not following http4s-core development closely, 0.16 is the last scalaz-stream based release.)

**Comment by zarthross** [comment]
I should think, Rho releases be at least synced up to the minor version of Http4s, even if the patch releases are out of sync.  

I would be happy to do another PR for a Http4s 0.16.* release of Rho but Rho is already released as 0.16 tied to 0.15.* of Http4s.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: scala
 scala:
 - 2.11.8
```
Comment (ChristopherDavenport) [inline .travis.yml]: Should have 2.11.11

File: core/src/test/scala/org/http4s/rho/bits/ResponseGeneratorSpec.scala [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
-
+import scodec.bits.ByteVector
 
 class ResponseGeneratorSpec extends Specification {
   import ResponseGeneratorInstances._
 
   "ResponseGenerator" should {
     "Build a response with a body" in {
-      val result = Ok("Foo").run
+      val result = Ok("Foo").unsafeRun
```
Comment (ChristopherDavenport) [inline core/src/test/scala/org/http4s/rho/bits/ResponseGeneratorSpec.scala]: Could this be a property test with an Arbitrary String? If not, some code smells can get fixed up after this is merged.

Comment (zarthross) [inline core/src/test/scala/org/http4s/rho/bits/ResponseGeneratorSpec.scala]: IMHO, thats out of the scope of this PR, the purpose was to get Rho updated for Http4s.  I agree we need to clean up A LOT of the tests, but I didn't want to change any of the behaviors of the code in this PR.

Comment (ChristopherDavenport) [inline core/src/test/scala/org/http4s/rho/bits/ResponseGeneratorSpec.scala]: Totally fair. Ill make a memo to myself to clean these up later

**Comment by bryce-anderson** [comment]
I'm in the middle of looking, and am being pulled away, but I'll try to finish it tomorrow. Looks great so far @zarthross! I'm pleasantly surprised how little fallout there is, but maybe that is all in the last half. :)

**Comment by etorreborre** [comment]
That looks good to me too. When there's a RC version of both http4s / rho with fs2/cats I will do a pass on upgrading our libraries and will report back.

**Comment by zarthross** [comment]
Well... looks like fs2 is going to be migrating to cats-effect instead of Task (https://github.com/functional-streams-for-scala/fs2/blob/b08d08b7c680c8f376ef71ddfa8921685a249c72/docs/migration-guide-0.10.md)

So, looks like this work will have to be redone for FS2 v0.10.

**Comment by bryce-anderson** [comment]
Looks like we're good to go here. How do we want to manage the release of this? I suspect merge this PR and consider versioning later?

**Comment by etorreborre** [comment]
@bryce-anderson given the comment about the next fs2 version I think this should wait until fs2 gets moved to cats-effect.

**Comment by bryce-anderson** [comment]
sgtm.

**Comment by etorreborre** [comment]
Ok, I see, this is is following up on a normal http4s release anyway, there's on reason not to do it. So let's go.

**Comment by eklavya** [comment]
Please make a release with this?

**Comment by bryce-anderson** [comment]
Sorry, I'm away from civilization for another day or so but I'll get to it
soon unless someone beats me too it.

**Comment by ZizhengTai** [comment]
A realease would be really helpful.

**Comment by bryce-anderson** [comment]
@zarthross @etorreborre are we still planning on holding this until an official http4s/fs2 release?

**Comment by etorreborre** [comment]
@bryce-anderson no, I think we should release.

**Comment by zarthross** [comment]
I also think we should release it, but probably as a snapshot or milestone in line with the http4s version.

**Comment by bryce-anderson** [comment]
Okay, I'll see what needs to be done this evening and try and get a release
out.

**Comment by bryce-anderson** [comment]
This has been released as v0.17.0-M3. Should be in central soon.

**Pull request merged**